tests/simple/testdata/parse.textproto

```diff
@@ -174,3 +174,478 @@ section {
+    expr: "a.continue() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "else"
+    description: "Check that `else` can be used as a receiver function."
+    expr: "a.else() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "for"
+    description: "Check that `for` can be used as a receiver function."
+    expr: "a.for() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "function"
+    description: "Check that `function` can be used as a receiver function."
+    expr: "a.function() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "if"
+    description: "Check that `if` can be used as a receiver function."
+    expr: "a.if() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "import"
+    description: "Check that `import` can be used as a receiver function."
+    expr: "a.import() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "let"
+    description: "Check that `let` can be used as a receiver function."
+    expr: "a.let() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "loop"
+    description: "Check that `loop` can be used as a receiver function."
+    expr: "a.loop() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "package"
+    description: "Check that `package` can be used as a receiver function."
+    expr: "a.package() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "namespace"
+    description: "Check that `namespace` can be used as a receiver function."
+    expr: "a.namespace() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "return"
+    description: "Check that `return` can be used as a receiver function."
+    expr: "a.return() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "var"
+    description: "Check that `var` can be used as a receiver function."
+    expr: "a.var() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "void"
+    description: "Check that `void` can be used as a receiver function."
+    expr: "a.void() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+  test {
+    name: "while"
+    description: "Check that `while` can be used as a receiver function."
+    expr: "a.while() || true"
+    disable_check: true
+    value { bool_value: true }
+  }
+}
+section {
+  name: "struct_names"
+  description: "Check that reserved identifiers are permitted as struct names as long as they are not language keywords"
+  test {
+    name: "as"
+    description: "Check that `as` can be used as a struct name."
+    expr: "as{} || true"
```

Comment:
Running these tests in cel-go at d8351df yields the error "unknown type: as" when planning the program ([conformance_test.go:237](https://github.com/google/cel-go/blob/d8351dfc77f4de9ad68ec7c95f8b342fcc3d6068/conformance/conformance_test.go#L237)).

So the expressions parse sucessfully, confirming that the identifiers are permitted in cel-go, but the planner expects a type `as` to be declared.